src/types.rs

```diff
@@ -63,6 +65,25 @@ impl<A> TypeInner<A> {
             },
             TypeInner::Boolean => f.write_str("bool"),
             TypeInner::UInt(ty) => write!(f, "{ty}"),
+            TypeInner::Tuple(elements) => match n_children_yielded {
+                0 => {
+                    f.write_str("(")?;
+                    if 0 == elements.len() {
+                        f.write_str(")")?;
+                    }
+                    Ok(())
+                }
+                n if n == elements.len() => {
+                    if n == 1 {
```

Comment:
In a01cfa130f0bab5d370bf9781bc133ca05d10133:

I think this should be `if n != 0`.

Reply:
The case `n == 1` is for printing the trailing comma in `(1,)` as opposed to printing `(1)`. Normal (nontrailing) commas as in `(1, 2)` are printed in the match arm `n` where `n != elements.len()`.